server/app/assets/javascripts/session.ts

```diff
@@ -59,10 +59,47 @@ export class SessionTimeoutHandler {
   private static hasTotalLengthWarningBeenShown = false
   /** Current active timeout timer */
   private static timer: number | null = null
+  /** Tracks if handler has been initialized */
+  private static isInitialized = false
+
+  /** Stores the next timeout action and time for testing */
+  private static nextTimeoutAction: (() => void) | null = null
+  private static nextTimeoutTime: number | null = null
 
   static init() {
+    if (this.isInitialized) {
+      return
+    }
+
     void this.checkAndSetTimer()
     this.setupModalEventHandlers()
+    // Add listener for timechange event to support testing with mocked time
```

Comment:
suggestion (non-blocking): Make it extra obvious this logic is used only for testing by factoring it out into a helper method and adding a method comment that it is used only for testing.